app/pipeline_factory.cpp

```diff
@@ -77,6 +91,110 @@ PipelineFactory::getReader(const op::variables_map &options) const {
   return nullptr;
 }
 
+std::vector<std::unique_ptr<FrameWindowFiltering>>
+PipelineFactory::getWindowFilters(const op::variables_map &options) const {
+  std::vector<std::unique_ptr<FrameWindowFiltering>> filters;
+  if (options.count("pipeline-frame-window-filtering") == 0) {
+    return filters;
+  }
+  std::vector<std::string> targets =
+      options["pipeline-frame-window-filtering"].as<std::vector<std::string>>();
+  for (const auto &target : targets) {
+    auto ptr = getWindowFiltering(target, options);
+    if (ptr.get() == nullptr) {
+      continue;
+    }
+    filters.push_back(std::move(ptr));
+  }
+  return filters;
+}
+
+DisparityMorphology::KernelShape morphShapeToEnum(const std::string &shapeStr) {
+  DisparityMorphology::KernelShape shape;
+  if (shapeStr == "rect") {
+    shape = DisparityMorphology::KernelShape::rect;
+  } else if (shapeStr == "ellipse") {
+    shape = DisparityMorphology::KernelShape::ellipse;
+  } else if (shapeStr == "cross") {
+    shape = DisparityMorphology::KernelShape::cross;
+  } else {
+    BOOST_LOG_TRIVIAL(warning)
+        << "Unknown disparity-morphology-shape: " << shapeStr
+        << ". Falling back to rect.";
+    shape = DisparityMorphology::KernelShape::rect;
+  }
+  return shape;
+}
+
+std::unique_ptr<FrameWindowFiltering>
+PipelineFactory::getWindowFiltering(const std::string &target,
+                                    const op::variables_map &options) const {
+  if (target == "disparity-gauss") {
+    auto ptr =
+        std::unique_ptr<DisparityGaussianBlur>(new DisparityGaussianBlur());
+    int k = options["disparity-gauss-k"].as<int>();
+    double sigma;
+    if (options.count("disparity-gauss-sigam")) {
+      sigma = options["disparity-gauss-sigma"].as<double>();
+    } else {
+      sigma = 0.3 * k + 0.8;
```

Comment:
What is this? Explain maybe

Reply:
good point, I'll add a comment. In short: they come from OpenCV, I haven't thought about the exact meaning, but I guess it's something like "make sure, 90% of the guassian weight is inside the window". At least that's what `0.3*k` suggests to me. `+0.8` might be some lower bound.